test/backend/range.in.cpp

```diff
@@ -81,3 +82,46 @@ NGRAPH_TEST(${BACKEND_NAME}, range)
         ASSERT_EQ(results, test.expected_result);
     }
 }
+
+NGRAPH_TEST(${BACKEND_NAME}, range_subgraph)
+{
+    // Create a graph for f(start,stop,step) = Range(start,stop,step).
+    auto start = make_shared<op::Parameter>(element::i32, Shape{});
+    auto stop = make_shared<op::Parameter>(element::i32, Shape{});
+    auto step = make_shared<op::Parameter>(element::i32, Shape{});
+    PartialShape out_max_shape{15};
+
+    auto range = make_shared<op::Range>(start, stop, step);
+    range->output(0).set_max_partial_shape(out_max_shape);
+
+    auto negative = make_shared<op::Negative>(range);
+
+    auto f = make_shared<Function>(NodeVector{negative}, ParameterVector{start, stop, step});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}", true);
+
+    auto handle = backend->compile(f);
+    pass::Manager passes;
+    passes.register_pass<pass::MinMaxShapePropagation>();
+    passes.run_passes(f);
+
+    auto t_start = backend->create_tensor(element::i32, Shape{});
+    copy_data(t_start, vector<int32_t>{0});
+    auto t_stop = backend->create_tensor(element::i32, Shape{});
+    copy_data(t_stop, vector<int32_t>{10});
+    auto t_step = backend->create_tensor(element::i32, Shape{});
+    copy_data(t_step, vector<int32_t>{1});
+    auto result = backend->create_dynamic_tensor(element::i32, PartialShape::dynamic());
```

Comment:
@sayantan-nervana see this unit-test on how max shape is propagated , and can be used by backend. We will need to explore how to use executable->create_outpu_tensor here instead of create_dynamic_tensor (not sure TF bridge uses this).